bokehjs/src/lib/embed/standalone.ts

```diff
@@ -44,16 +47,18 @@ export async function add_document_standalone(document: Document, element: HTMLE
     }
   }
 
-  for (const model of document.roots())
-    await render_model(model)
+  const root_models = document.roots()
+  for (let i = 0; i < root_models.length; i++)
+    await render_model(root_models[i], i)
 
   if (use_for_title)
     window.document.title = document.title()
 
   document.on_change((event: DocumentChangedEvent): void => {
-    if (event instanceof RootAddedEvent)
-      render_model(event.model)
-    else if (event instanceof RootRemovedEvent)
+    if (event instanceof RootAddedEvent) {
+      const count = document.roots().indexOf((event.model as any))
```

Comment:
Would prefer to not use `as any` here but it was complaining that event.model was `HasProps` typed while `document.roots()` is `Model[]` typed.

Reply:
That's a common problem. The `Model` and `HasProps` dichotomy across bokehjs is a mess right now.